addons/httprox/common/src/main/java/org/commonjava/indy/httprox/util/HttpConduitWrapper.java

```diff
@@ -227,6 +231,8 @@ public void writeExistingTransfer( Transfer txfr, boolean writeBody, String path
         {
             cacheProvider.cleanupCurrentThread();
         }
+        sinkChannel.flush();
```

Comment:
I wonder if this might have been what was keeping the server from sending a FIN response to the client last week when proxying in non-MITM mode?

Reply:
I am not sure, but this may help. I remember there is a close method which also calls the flush. Let's see what is happening after merging this.